components/fxa-client/src/telemetry.rs

```diff
@@ -0,0 +1,13 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pub use crate::http_client::ProfileResponse as Profile;
+use crate::{error::*, scopes, util, CachedResponse, FirefoxAccount};
+
+impl FirefoxAccount {
+    pub fn get_ecosystem_anon_id(&mut self) -> Result<Option<String>> {
+        let profile = self.get_profile(false)?;
```

Comment:
Note that I am not storing the ID in the FxA state separately, which should not matter? (well, it might for offline cases maybe). I'm happy to move it there if you think it's valuable.

Reply:
This seems fine, we'll probably revisit state management a bit in #3086 but can cross that bridge when we get there.